hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerManager.java

```diff
@@ -126,6 +131,113 @@ public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerStatus(
     }
   }
 
+  /**
+   * Send startDiskBalancer command to datanodes.
+   * If hosts is not specified, send commands to all healthy datanodes.
+   * @param threshold new configuration of threshold
+   * @param bandwidthInMB new configuration of bandwidthInMB
+   * @param parallelThread new configuration of parallelThread
+   * @param hosts Datanodes that command will apply on
+   * @return Possible errors
+   * @throws IOException
+   */
+  public List<DatanodeAdminError> startDiskBalancer(
+      Optional<Double> threshold, Optional<Long> bandwidthInMB,
+      Optional<Integer> parallelThread, Optional<List<String>> hosts)
+      throws IOException {
+    List<DatanodeDetails> dns;
+    if (hosts.isPresent()) {
+      dns = NodeUtils.mapHostnamesToDatanodes(nodeManager, hosts.get(),
+          useHostnames);
```

Comment:
Should avoid send command to dn which is not healthy.

Reply:
Updated, please have a check.